core/io/image.cpp

```diff
@@ -516,6 +516,7 @@ void Image::convert(Format p_new_format) {
 	if (p_new_format == format) {
 		return;
 	}
+	ERR_FAIL_INDEX_MSG(p_new_format, FORMAT_MAX, "The Image format specified (" + itos(p_new_format) + ") is out of range. See Image's Format enum.");
```

Comment:
Generally speaking we prefer validating input as early as possible. So this check should be moved to the top of the method, since its validity does not depend on other checks passing.

Reply:
I moved it above the other checks. Is there anything else I should know about this? I appreciate all of your helpful feedback so far.